astroid/brain/brain_builtin_inference.py

```diff
@@ -641,12 +646,13 @@ def infer_property(
 
     prop_func = objects.Property(
         function=inferred,
-        name=inferred.name,
+        name="<property>",
```

Comment:
Why this change? Doesn't that lose value?

Reply:
Right, I talk about it in the commit message:

- fixing construction of in-place properties (`bar =
   property(getter)`). They just create a nameless object, not the one
   with the name of the getter. Thus, the name was changed to
   "<property>". Furthermore, the definition of that property is not
   attached to any scope, as it's again nameless.

This change is for "inplace", nameless, properties, not the ones created with a decorator. 

What does it mean for it to lose the value? It's still accessible through an `Assign` field, for example.

Reply:
Ah, makes sense. I understand the name change a little better now (although I would probably do `f"<property> {inferred.name}"` or something, so we don't lose that context.

I don't really understand the `parent` changes though.

```python
class A:
    temperature = property(get_temperature, set_temperature)
```

For `temperature` I would still expect `A` to be the `parent` no? Not the `adhoc` module

Reply:
For `temperature` it is. However, temperature is just a name in the AssignName. It is assigned a call to the constructor of actual `property`. The actual `property` class is defined somewhere in the builtins module. However, since we are treating `property` specially, we are creating our own definition of it, separate for each use. Kind of like an Instance. And that special definition isn't defined anywhere. We've just made it up on the spot, ad-hoc. 

Treating it as if it was defined in `A` is furthermore incorrect, since it would cause it to be in `A`s body, or searchable from `A`.

Reply:
I understand now, thanks!